src/main/java/io/getunleash/repository/JsonFeatureParser.java

```diff
@@ -0,0 +1,29 @@
+package io.getunleash.repository;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.Reader;
+
+final class JsonFeatureParser {
+
+    private JsonFeatureParser() {}
+
+    public static String toJsonString(FeatureCollection featureCollection) {
+        Gson gson = new GsonBuilder().create();
```

Comment:
It's been a long time since I've worked with GSON but if memory serves, the GSON objects are both thread safe and expensive to instantiate. Might be worthwhile to make these once and hold them if possible